app/api/chemotion/chem_spectra_api.rb

```diff
@@ -201,14 +201,10 @@ def raw_file(att)
         route_param :host_name do
           get do
             if Rails.configuration.spectra.nmriumwrapper.blank?
-              { protocol: '', host: '', port: '' }
+              { nmrium_url: '' }
             else
               nmrium_url = Rails.configuration.spectra.nmriumwrapper.url
-              nmrium_uri = URI(nmrium_url)
-              protocol = nmrium_uri.scheme
-              host = nmrium_uri.host
-              port = nmrium_uri.port == 443 ? '' : nmrium_uri.port
-              { protocol: protocol, host: host, port: port }
+              { nmrium_url: nmrium_url }
             end
```

Comment:
Starting from 203 following, couldn't this simply be:

```rb
{ nmrium_url: Rails.configuration.spectra.nmriumwrapper.url.presence || '' }
```

Maybe even without the `|| ''`?

Reply:
@ptrxyz I updated